cli/src/commands/operation/log.rs

```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::slice;
+
 use jj_lib::op_walk;
 
 use crate::cli_util::{format_template, CommandHelper, LogContentFormat};
```

Comment:
Perhaps add some docs:

> Show the operation log
>
> Like other `jj` operations, by default `jj op log` may snapshot the working copy or merge most recent concurrent operations. If you'd like to make sure the operation is read-only, use `jj op log --at-op @ --ignore-working-copy`.

Also, if there is concurrency, it seems there is no way to see both concurrent operations in on `jj op log` without merging them, is there? It would be nice if `jj op log --at-op @` showed both (I guess it'd happen at the auto-merge of the concurrent operations without recording the merge), but I'm guessing now it'd be an error message (or did you figure out a better solution?).

If that's correct, I suppose we could add something like

> If there is no unique latest operation, `jj op log --at-op @` will produce an informative error message that gives ids for the most recent concurrent operations.

Reply:
I think the goal is to remove special case and docs about the special case (such as `op log` doesn't snapshot but `op diff` does.)

Maybe we can add more detailed hint if `op log --at-op=@` failed to load the concurrent heads, but I assume the situation would be pretty rare.

Reply:
> Maybe we can add more detailed hint if op log --at-op=@ failed to load the concurrent heads

I think that would be a fine alternative. If `--at-op=@` does not imply `--ignore-working-copy`, that might need a detailed hint too. Ultimately, I'll leave this up to you as well, but here's my motivation.

My main concern is that we might want to pay special attention to commands used to debug or recover from broken states. Knowing which commands are read-only can be critical in this case.

More philosophically, I think that beginner or intermediate `jj` users (especially early adopters) would often get into broken states. We should be motivated to help them, not just out of benevolence, but also because we disproportionately benefit from bug reports such users provide. So, it's in our interest to give them all the guidance we can on how to get meaningful debug info and how to recover. But I may be biased, since I tend to put myself into weird situations often.

Reply:
Update the help text. Can you take another look?
```
/// Like other commands, `jj op log` snapshots the current working-copy changes
/// and merges concurrent operations. Use `--at-op=@ --ignore-working-copy` to
/// inspect the current state without mutation.
```

Reply:
This text looks good to me, thank you!

Reply:
Although, it still bothers me that if the merge of concurrent operations is failing, there's no way to see both concurrent operations at once, since `--at-op @` will give an error. Hopefully that would be rare?

Reply:
> Hopefully that would be rare?

I believe it is. Typical use case is `op log` or `op log --ignore-working-copy`, and concurrent ops get merged. `--at-op=@` is needed only when merge fails (i.e. repository is somewhat corrupted.)